File: Queues/ExerciseBuildingPriorityQueues/priorityQueues.py
class PriorityQueue:
    def __init__(self):
        self.__qArray = []

    def enqueue(self, element):
        if (len(self.__qArray) == 0):
            return self.__qArray.append(element)

        for i in range(len(self.__qArray)):
            if (self.__qArray[i] <= element):
                return self.__qArray.insert(len(self.__qArray), element)

    def getQueue(self):
        if (len(self.__qArray) == 0):
            raise Exception('The queue is empty')

        return self.__qArray


queue = PriorityQueue()
queue.enqueue(5)
queue.enqueue(1)
queue.enqueue(3)

print(queue.getQueue())

File: Queues/ExerciseBuildingPriorityQueues/priorityQueues.js
const _qArray = new WeakMap();

class PriorityQueue{
    constructor(){
        _qArray.set(this, [])
    }

    enqueue(element){
        if (_qArray.get(this).length == 0) 
            return _qArray.get(this).push(element);
    
        for (let i = _qArray.get(this).length; i > 0; i--){
            if (_qArray.get(this)[i-1] <= element) 
                return _qArray.get(this).splice(i, 0, element);   
        }
    }

    dequeue(){
        if (_qArray.get(this).length == 0) throw new Error('The queue is empty');
        return _qArray.get(this).shift();
    }

    peek(){
        if (_qArray.get(this).length == 0) throw new Error('The queue is empty');
        return _qArray.get(this)[0];
    }

    isEmpty(){
        return (_qArray.get(this).length == 0);
    }

    getQueue(){
        return _qArray.get(this)
    }

}

const queue = new PriorityQueue();

queue.enqueue(1);
queue.enqueue(3);
queue.enqueue(2);
console.log(queue.getQueue());